src/components/Pressable.tsx

```diff
@@ -0,0 +1,277 @@
+});
+
+const isTouchWithinInset = (
+  // all of these are guaranteed to be their type,
+  // but we cannot use '!' anymore
+  touch?: TouchData,
+  inset?: Insets,
+  dimensions?: { width: number; height: number }
+) =>
+  // ^ to right, ^ to bottom
+  (touch?.x ?? 0) < (inset?.right ?? 0) + (dimensions?.width ?? 0) &&
+  (touch?.y ?? 0) < (inset?.bottom ?? 0) + (dimensions?.height ?? 0) &&
+  (touch?.x ?? 0) > -(inset?.left ?? 0) &&
+  (touch?.y ?? 0) > -(inset?.top ?? 0);
+
+const adaptStateChangeEvent = (
+  event: GestureStateChangeEvent<
+    HoverGestureHandlerEventPayload | LongPressGestureHandlerEventPayload
+  >
+): PressableEvent => {
+  const timestamp = Date.now();
+
+  const touchData = changeToTouchData(event);
+
+  const pressEvent = touchToPressEvent(touchData, timestamp);
+
+  return {
+    nativeEvent: {
+      touches: [pressEvent],
+      changedTouches: [pressEvent],
+      identifier: event.handlerTag,
+      locationX: event.x ?? -1,
+      locationY: event.y ?? -1,
+      pageX: event.absoluteX ?? -1,
+      pageY: event.absoluteY ?? -1,
+      target: 0, // node ID
+      timestamp: timestamp,
+      force: undefined,
+    },
+  };
+};
+
+const adaptTouchEvent = (event: GestureTouchEvent): PressableEvent => {
+  const timestamp = Date.now();
+
+  const nativeTouches = event.allTouches.map((touch) =>
+    touchToPressEvent(touch, timestamp)
+  );
+  const nativeChangedTouches = event.changedTouches.map((touch) =>
+    touchToPressEvent(touch, timestamp)
+  );
+
+  return {
+    nativeEvent: {
+      touches: nativeTouches,
+      changedTouches: nativeChangedTouches,
+      identifier: event.handlerTag,
+      locationX: event.allTouches.at(0)?.x ?? -1,
+      locationY: event.allTouches.at(0)?.y ?? -1,
+      pageX: event.allTouches.at(0)?.absoluteX ?? -1,
+      pageY: event.allTouches.at(0)?.absoluteY ?? -1,
+      target: 0, // node ID
+      timestamp: timestamp,
+      force: undefined,
+    },
+  };
+};
+
+export default function Pressable(props: PressableProps) {
+  const pressableRef = useRef<View>(null);
+  const [styleProp, setStyleProp] = useState<
+    StyleProp<ViewStyle> | undefined
+  >();
+  const [childrenProp, setChildrenProp] = useState<
+    React.ReactNode | undefined
+  >();
+
+  // disabled when onLongPress has been called
+  const isPressCallbackEnabled = useRef<boolean>(true);
+  const isPressedDown = useRef<boolean>(false);
+
+  const normalizedHitSlop: Insets =
+    typeof props.hitSlop === 'number'
+      ? numberAsInset(props.hitSlop)
+      : props.hitSlop ?? {};
+
+  const normalizedPressRetentionOffset: Insets =
+    typeof props.pressRetentionOffset === 'number'
+      ? numberAsInset(props.pressRetentionOffset)
+      : props.pressRetentionOffset ?? {};
+
+  const pressGesture = Gesture.LongPress().onStart((event) => {
+    if (isPressedDown.current) {
+      props.onLongPress?.(adaptStateChangeEvent(event));
+      isPressCallbackEnabled.current = false;
+    }
+  });
```

Comment:
It may be worthwhile to consider wrapping gesture objects with `useMemo` so that they are not recreated on every render.

Reply:
Done